modules/repository/variables.tf

```
@@ -0,0 +1,57 @@
+variable "name" {}
+
+variable "supermarket_name_override" {
+  default = ""
+}
+
+variable "repo_type" {
+  type = string
+  validation {
+    condition     = can(regex("^cookbook|terraform|ide|other$", var.repo_type))
+    error_message = "The repo_type must be cookbook, terraform, ide or other. Case Sensitive."
+  }
+}
+variable "description" {
+  type    = string
+  default = ""
+}
+variable "homepage_url" {
+  type    = string
+  default = ""
+}
+variable "topics" {
+  type    = list(string)
+  default = []
+}
+
+locals {
+  // supermarket_name
+  supermarket_name = var.supermarket_name_override == null ? var.name : var.supermarket_name_override
+}
+
+locals {
+  // topics only
+  default_topics = ["terraform-managed"]
+
+  chef_topics       = var.repo_type == "cookbook" ? ["chef", "chef-cookbook", "chef-resource", "${replace(replace(local.supermarket_name, "_", "-"), ".", "")}", "hacktoberfest"] : []
+  ide_topics        = var.repo_type == "ide" ? ["ide", "${replace(replace(var.name, "_", "-"), ".", "")}"] : []
+  terraform_topics  = var.repo_type == "terraform" ? ["terraform", "${replace(replace(var.name, "_", "-"), ".", "")}"] : []
+  additional_topics = var.repo_type != "cookbook" && var.topics != null ? var.topics : []
+  topics            = concat(local.default_topics, local.chef_topics, local.ide_topics, local.terraform_topics, local.additional_topics)
```

```suggestion
  topics            = compact(concat(local.default_topics, local.chef_topics, local.ide_topics, local.terraform_topics, local.additional_topics))
```

You'll cant to [compact this to get rid of the empty items](https://www.terraform.io/docs/language/functions/compact.html)